app.js

```diff
@@ -1,13 +1,29 @@
 const express = require("express");
 const app = express();
 
-const { listArticles, listTopics, handle404Errors } = require("./controllers/controller")
+const { 
+    listArticles, 
+    listTopics, 
+    findSpecificArticle 
+} = require("./controllers/controller")
 
 
 app.get("/api/topics", listTopics)
 
 app.get("/api/articles", listArticles)
 
-app.all("/*", handle404Errors)
+app.get("/api/articles/:article_id", findSpecificArticle)
 
-module.exports = app
\ No newline at end of file
+app.all("/*", (request, response, next) => {
+  response.status(404).send({ msg: "404 - Not found" });
+  next();
+});
+
+app.use((error, request, response, next) => {
+  if (error.code === "22P02" || error.msg === "404 - No rows found") {
+    response.status(404).send({ msg: "404 - Not found" });
+  }
+  next();
+});
+
```

Comment:
You are missing a custom error handler "handleCustomErrors" which sends back err.status and err.message. This will deal with your 404 instead 👍